exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -2233,3 +2168,62 @@ subSort:
 		break
 	}
 }
+
+// subToMap returns a json object of request params for subscriptions
+func subToMap(s *subscription.Subscription, a asset.Item, p currency.Pair) map[string]any {
+	c := s.Channel
+	if name, ok := subscriptionNames[s.Channel]; ok {
+		c = name
+	}
+	req := map[string]interface{}{
+		"channel": c,
+	}
+
+	var fundingPeriod string
+	for k, v := range s.Params {
+		switch k {
+		case CandlesPeriodKey:
+			if s, ok := v.(string); !ok {
+				panic(common.GetTypeAssertError("string", v, "subscription.CandlesPeriodKey"))
+			} else {
+				fundingPeriod = ":" + s
+			}
+		case "key", "symbol", "len":
+			panic(fmt.Errorf("%w: %s", errParamNotAllowed, k)) // Ensure user's Params aren't silently overwritten
```

Comment:
FuncMaps can support errors. Yes it makes things less neat, but I don't think panics are appropriate to introduce. The templating work already allows for errors to be printed

Reply:
panic isn't really a panic. It's caught by text/template and thrown as a nice error.

Fixed 379dd7145 by documentation

Reply:
```
[ERROR] | WEBSOCKET | 14/10/2024 12:54:02 | Bitfinex websocket: subscription failure, ticker all : template: master.tmpl:5:6: executing "master.tmpl" at <subToMap $.S $asset $p>: error calling subToMap: param not allowed: ticker all , allTrades all : template: master.tmpl:5:6: executing "master.tmpl" at <subToMap $.S $asset $p>: error calling subToMap: param not allowed: allTrades all , candles spot : template: master.tmpl:5:6: executing "master.tmpl" at <subToMap $.S $asset $p>: error calling subToMap: param not allowed: candles spot , orderbook all : template: master.tmpl:5:6: executing "master.tmpl" at <subToMap $.S $asset $p>: error calling subToMap: param not allowed: orderbook all 
```
Confirming that is the case with my own panic